clusterloader2/pkg/imagepreload/imagepreload.go

```diff
@@ -156,19 +157,37 @@ func (c *controller) PreloadImages() error {
 	return nil
 }
 
-func (c *controller) checkNode(obj interface{}, done map[string]bool) {
-	if obj == nil {
+func (c *controller) checkNode(set map[string]struct{}, old, new interface{}) {
+	if new != nil {
+		node := new.(*v1.Node)
+		preloaded := c.hasPreloadedImages(node)
+		c.markDone(set, node.Name, preloaded)
 		return
 	}
+	if old != nil {
```

Comment:
Not sure if I follow this line: I understand that this should catch case when node has been deleted (as you remove it from the map).

Why is it a case?
Don't we simply get checkNode([old version of node], [new version of node]) called for all the updates?

Please correct or add a comment why it is correct :)

Reply:
I missed "return" above. I think it's correct then, as this if is correct when new == nil && old != nil, which is a case for deletion.